bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingFactory.java

```diff
@@ -105,9 +108,13 @@ public static Thing createThing(ThingUID thingUID, Configuration configuration,
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                if (properties != null) {
-                    for (String key : properties.keySet()) {
-                        thing.setProperty(key, properties.get(key));
+                if (thing == null) {
+                    logger.error("Thing factory returned null when it reports to support the thing");
```

Comment:
Perhaps we could also add the thing handler factory class and all the UIDs.
What do you think, IMHO this more verbose message will identify the handler easier.